sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1190,7 +1189,9 @@ public Node visitJoinRelation(SqlBaseParser.JoinRelationContext context) {
                 criteria = new JoinOn((Expression) visit(context.joinCriteria().booleanExpression()));
             } else if (context.joinCriteria().USING() != null) {
                 List<String> columns = identsToStrings(context.joinCriteria().ident());
-                criteria = new JoinUsing(columns);
+                Table leftTable = (Table) left;
```

Comment:
Is it really safe to cast to `Table` here? What if `left` or `right` will be a sub query?

Reply:
Not safe, I will now add tests, like the one https://github.com/crate/crate/issues/9551 , which will highlight in catastrophic form exactly this :)